modules/apps/27-interchain-accounts/host/client/cli/tx_test.go

```diff
@@ -0,0 +1,155 @@
+package cli
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/require"
+
+	icatypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/types"
+)
+
+const msgDelegateMessage = `{
+	"@type": "/cosmos.staking.v1beta1.MsgDelegate",
+	"delegator_address": "cosmos15ccshhmp0gsx29qpqq6g4zmltnnvgmyu9ueuadh9y2nc5zj0szls5gtddz",
+	"validator_address": "cosmosvaloper1qnk2n4nlkpw9xfqntladh74w6ujtulwnmxnh3k",
+	"amount": {
+		"denom": "stake",
+		"amount": "1000"
+	}
+}`
+
+const bankSendMessage = `{
+	"@type":"/cosmos.bank.v1beta1.MsgSend",
+	"from_address":"cosmos15ccshhmp0gsx29qpqq6g4zmltnnvgmyu9ueuadh9y2nc5zj0szls5gtddz",
+	"to_address":"cosmos10h9stc5v6ntgeygf5xf945njqq5h32r53uquvw",
+	"amount": [
+		{
+			"denom": "stake",
+			"amount": "1000"
+		}
+	]
+}`
+
+var multiMsg = fmt.Sprintf("[ %s, %s ]", msgDelegateMessage, bankSendMessage)
+
+func TestGeneratePacketData(t *testing.T) {
+	tests := []struct {
+		name                string
+		memo                string
+		expectedPass        bool
+		message             string
+		registerInterfaceFn func(registry codectypes.InterfaceRegistry)
+		assertionFn         func(t *testing.T, msgs []sdk.Msg)
+	}{
+		{
+			name:         "multi message",
+			memo:         "",
+			expectedPass: true,
+			message:      multiMsg,
+			registerInterfaceFn: func(registry codectypes.InterfaceRegistry) {
+				stakingtypes.RegisterInterfaces(registry)
+				banktypes.RegisterInterfaces(registry)
+			},
+			assertionFn: func(t *testing.T, msgs []sdk.Msg) {
+				assertMsgDelegate(t, msgs, 0)
```

Comment:
Maybe a nit, but Instead of passing the slice of messages and the index, couldn't you just pass `msgs[0]`?

Reply:
Ah yes you're right, this can simplify things a little bit!